Framework/Interaction/Tax.php

```diff
@@ -921,4 +921,24 @@ protected function isProductCalculated($item) {
         }
         return false;
     }
+
+    /**
+     * This method will attempt to retrieve the provided customer code value as a system-defined customer attribute; if
+     * that fails, then it will attempt to retrieve the value as a custom attribute
+     *
+     * @param \Magento\Customer\Api\Data\CustomerInterface $customer
+     * @param string $customerCode
+     * @return mixed
+     */
+    protected function retrieveCustomerCode($customer, $customerCode)
+    {
+        // Convert provided customer code to getter name
+        $getCustomerCode = 'get' . str_replace('_', '', ucwords($customerCode, '_'));
+        if (method_exists($customer, $getCustomerCode)) {
```

Comment:
Could you use `$customer->getData($customerCode)` here instead?

Reply:
No, a call to $customer->getData() results in a "call to undefined method" exception.